Makefile

```diff
@@ -234,6 +234,11 @@ archi: $(EXTRA_DIR)/ocamldot/ocamldot
 	echo "}" >> archi.dot
 	dot -Tpdf archi.dot > archi.pdf
 
+lock: clean lib
```

Comment:
Why do we need `clean` and `lib` dependencies?

Reply:
I clean the _build directory to be sure the binary compile with the actual opam switch and I add the lib to check if the binary compile as we ask me. The command `dune build ./alt-ergo-lib.opam` doesn't exist.

Reply:
I only suggested to make sure that `alt-ergo-lib.opam` is up-to-date before locking it because it seemed sensible. Making sure that the library compiles makes sense as well, although I am not sure either of these things are necessary, I'm fine having them or not as you prefer.

However we shouldn't clean the build directory without a reason, and I don't see any reason to do it here -- if we want to check that the library builds, we can just do that, without clearing the build cache.

Not sure what you mean by "The command `dune bulid ./alt-ergo-lib.opam` doesn't exist." however, it should rebuild alt-ergo-lib.opam from the `dune-project`.

```shell
$ file alt-ergo-lib.opam
alt-ergo-lib.opam: cannot open `alt-ergo-lib.opam' (No such file or directory)

$ dune build ./alt-ergo-lib.opam
                                  
$ file alt-ergo-lib.opam
alt-ergo-lib.opam: ASCII text
```

Reply:
In fact the CI will fail in any case. So we can only rebuild the opam file ;)